File: app/src/main/java/com/example/marvelworld/ui/MarvelListViewModel.java
package com.example.marvelworld.ui;

import com.example.marvelworld.data.MarvelDataRepository;
import com.example.marvelworld.model.MarvelCharacter;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

public class MarvelListViewModel extends ViewModel {

    private final LiveData<List<MarvelCharacter>> marvelCharacters;
    private MarvelDataRepository repository;

    public MarvelListViewModel(String apiKey) {
        this.repository = new MarvelDataRepository(apiKey);
        this.marvelCharacters = repository.getMarvelCharacters();
    }

    public LiveData<List<MarvelCharacter>> getMarvelCharacters() {
        return marvelCharacters;
    }
}

File: app/src/main/java/com/example/marvelworld/ui/MarvelListViewModel.kt
package com.example.marvelworld.ui

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.marvelworld.data.MarvelDataRepository
import com.example.marvelworld.model.MarvelCharacter

class MarvelListViewModel(apiKey: String?) : ViewModel() {
    val marvelCharacters: LiveData<List<MarvelCharacter>>
    private val repository: MarvelDataRepository = MarvelDataRepository(apiKey!!)

    init {
        marvelCharacters = repository.getMarvelCharacters()
    }
}